smt/surrogate_models/kpls.py

```diff
@@ -100,33 +102,32 @@ def _estimate_number_of_components(self):
                 self.training_points[None][0][1] = yfold
                 # We are doing hyperparameters hot_start for the k-fold
                 # as the problem is the same and the data are mainly similar
-                if fold == 0:
-                    super(KPLS, self)._initialize()
-                else:
-                    super(KPLS, self)._initialize()
+                super(KPLS, self)._initialize()
+                self.options["corr"] = corr
```

Comment:
Hi Paul. Could you elaborate on this change? I understand that you've saved `corr` option and re-assign it here because the `_initialize` call re-declares options and resets it to the default value but in that case we should do the same thing with all options? Or did I miss something? Finally why do you call `_initialize` ? Maybe you should reset relevant fields only?